lib/front_desk.rb

```diff
@@ -0,0 +1,105 @@
+require_relative 'reservation'
+require_relative 'date_range'
+
+module Hotel
+  class FrontDesk
+    attr_accessor :reservations, :rooms, :calendar
+
+    def initialize
+      @rooms = (1..20).map { |i| i }
+      @reservations = []
+      @calendar = {}
+    end
+
+    def populate_calendar(reservation)
+      temp = reservation.date_range.dates
+      temp.pop
+      temp.each do |date|
+        if !@calendar.key?(date)
+          @calendar[date] =[]
+        end
+        @calendar[date] << reservation
+      end
+      return @calendar
+    end
```

Comment:
I think you probably meant to copy the array here. This doesn't do that so you wind up changing dates.

Cleaner would be to ask for a range that excludes the end of `dates` (you can also use an `unless` to tidy up your condition):

```suggestion
    def populate_calendar(reservation)
      reservation.date_range.dates[0...-1].each do |date|
        unless @calendar.key?(date)
          @calendar[date] = []
        end
        @calendar[date] << reservation
      end
      return @calendar
    end
```

Reply:
Also, since this method doesn't directly address a user story I'd recommend making it private.